sql/postgres/migrate.go

```diff
@@ -630,44 +630,49 @@ func (s *state) columnDefault(b *sqlx.Builder, c *schema.Column) {
 	}
 }
 
-func (s *state) alterColumn(b *sqlx.Builder, k schema.ChangeKind, c *schema.Column) error {
-	for !k.Is(schema.NoChange) {
-		b.P("ALTER COLUMN").Ident(c.Name)
+func (s *state) alterColumn(b *sqlx.Builder, c *schema.ModifyColumn) error {
+	for k := c.Change; !k.Is(schema.NoChange); {
+		b.P("ALTER COLUMN").Ident(c.To.Name)
 		switch {
 		case k.Is(schema.ChangeType):
-			t, err := FormatType(c.Type.Type)
+			t, err := FormatType(c.To.Type.Type)
 			if err != nil {
 				return err
 			}
 			b.P("TYPE", t)
-			if collate := (schema.Collation{}); sqlx.Has(c.Attrs, &collate) {
+			if collate := (schema.Collation{}); sqlx.Has(c.To.Attrs, &collate) {
 				b.P("COLLATE", collate.V)
 			}
 			k &= ^schema.ChangeType
-		case k.Is(schema.ChangeNull) && c.Type.Null:
+		case k.Is(schema.ChangeNull) && c.To.Type.Null:
 			b.P("DROP NOT NULL")
 			k &= ^schema.ChangeNull
-		case k.Is(schema.ChangeNull) && !c.Type.Null:
+		case k.Is(schema.ChangeNull) && !c.To.Type.Null:
 			b.P("SET NOT NULL")
 			k &= ^schema.ChangeNull
-		case k.Is(schema.ChangeDefault) && c.Default == nil:
+		case k.Is(schema.ChangeDefault) && c.To.Default == nil:
 			b.P("DROP DEFAULT")
 			k &= ^schema.ChangeDefault
-		case k.Is(schema.ChangeDefault) && c.Default != nil:
-			s.columnDefault(b.P("SET"), c)
+		case k.Is(schema.ChangeDefault) && c.To.Default != nil:
+			s.columnDefault(b.P("SET"), c.To)
 			k &= ^schema.ChangeDefault
 		case k.Is(schema.ChangeAttr):
-			id, ok := identity(c.Attrs)
+			toI, ok := identity(c.To.Attrs)
 			if !ok {
-				return fmt.Errorf("unexpected attribute change (expect IDENTITY): %v", c.Attrs)
+				return fmt.Errorf("unexpected attribute change (expect IDENTITY): %v", c.To.Attrs)
 			}
 			// The syntax for altering identity columns is identical to sequence_options.
 			// https://www.postgresql.org/docs/current/sql-altersequence.html
-			b.P("SET GENERATED", id.Generation, "SET START WITH", strconv.FormatInt(id.Sequence.Start, 10), "SET INCREMENT BY", strconv.FormatInt(id.Sequence.Increment, 10), "RESTART")
+			b.P("SET GENERATED", toI.Generation, "SET START WITH", strconv.FormatInt(toI.Sequence.Start, 10), "SET INCREMENT BY", strconv.FormatInt(toI.Sequence.Increment, 10))
+			// Skip SEQUENCE RESTART in case the "start value" is less than the "current value" in one
+			// of the states (inspected and desired), because this function is used for both UP and DOWN.
+			if fromI, ok := identity(c.From.Attrs); (!ok || fromI.Sequence.Current < toI.Sequence.Start) && toI.Sequence.Current < toI.Sequence.Start {
+				b.P("RESTART")
+			}
```

Comment:
This also means when changing the start value in the desired state, nothing will happen, correct? Maybe we should also add some kind of analyzer for this one, checking if there is an sequence change, that can possibly break behaviour.

Reply:
> This also means when changing the start value in the desired state, nothing will happen, correct?

Yes.